contracts/w3bstreaming/src/lib.rs

```diff
@@ -27,56 +27,115 @@ extern fn handle() {
             img_link,
         } => {
             let stream_id = exec::block_timestamp().to_string() + &title;
-            if let Some(profile) = contract.users.get_mut(&msg::source()) {
+            let msg_src = msg::source();
+            if let Some(profile) = program.users.get_mut(&msg_src) {
                 profile.stream_ids.push(stream_id.clone());
             } else {
                 panic!("Account is no registered");
             }
-            contract.streams.insert(
+            program.streams.insert(
                 stream_id.clone(),
                 Stream {
-                    broadcaster: msg::source(),
+                    broadcaster: msg_src,
                     img_link,
                     start_time,
                     end_time,
                     title,
                     description,
-                    watchers: Vec::new(),
                 },
             );
-            msg::reply(ActionResult::StreamIsScheduled { id: stream_id }, 0)
+            msg::reply(Event::StreamIsScheduled { id: stream_id }, 0)
                 .expect("Unable to send reply");
         }
+        Action::DeleteStream { stream_id } => {
+            let msg_src = msg::source();
+            if let Some(profile) = program.users.get_mut(&msg_src) {
+                if let Some(index) = profile.stream_ids.iter().position(|x| *x == stream_id) {
+                    profile.stream_ids.remove(index);
+                } else {
+                    panic!("Id is not exist");
+                }
+            } else {
+                panic!("Account is no registered");
+            }
```

Comment:
`if let Some(x) = y { ... } else { panic!("message") }` could be replaced by `let x = y.expect("message");`